### PR TITLE
aseprite-unfree: 1.2.16.3 -> 1.2.40 and drop the free version 

### LINKED ISSUE #128308: aseprite-unfree: package outdated
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: aseprite-unfree

current version: 1.2.16.3 (3 Dec 2019)
~desired version: 1.2.27 (19 Apr 2021)~
desired version: 1.2.30 (26 Oct 2021)

###### Notify maintainers

maintainers: @orivej 

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by kyren**
###### Description of changes

This PR contains two separate commits, the first updates the Aseprite unfree version to the latest released version: 1.2.40, the second removes the free version of Aseprite and aliases 'aseprite-unfree' to 'aseprite'. The motivation for this is that there is already a better free version of Aseprite in nixpkgs, namely the LibreSprite fork. The only reason somebody would want to install Aseprite is for the proprietary version.

In order to get 1.2.40 to build, I borrowed some patches from the AUR PKGBUILD:

https://aur.archlinux.org/cgit/aur.git/tree/?h=aseprite&id=3e063a0a020301b652d1fbea8dbe47cc9851c3d7

Only shared-libwebp.patch and shared-skia-deps.patch, just enough to get the package to actually build and function.

I'm still pretty new at hacking nixpkgs so there's a high likelihood this isn't perfect. There might be things that can be removed from the old derivation that are no longer necessary, and I didn't spend much time figuring out what those are, if any.

If for some reason we want to keep the free version of Aseprite, I can simply drop the second commit, but I can't see a compelling reason to keep it?

Closes: #128308

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---
Edit: I guess I was supposed to ping the maintainer? @orivej

### CODE REVIEW

**Comment by kyren** [comment]
Ah, I seem to have screwed up the desktop integration, fixing it..

File: pkgs/top-level/all-packages.nix
Comment (kyren) [reply pkgs/top-level/all-packages.nix:27297]: I can still do that, it doesn't seem like a terrible idea?

Comment (tjni) [reply pkgs/top-level/all-packages.nix:27297]: Feel free, your call :)

Comment (kyren) [reply pkgs/top-level/all-packages.nix]: I went ahead and moved the alias to top-level/aliases.nix and also added release notes

**Comment by kyren** [comment]
I also added release notes for 22.11, if it's possible for the update to be included (if not, I'll try and backport). I'm also happy to add myself as a maintainer since I use Aseprite quite a lot, I just have never maintained a package before.

File: nixos/doc/manual/release-notes/rl-2211.section.md
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2211.section.md:343]: This change must be moved to 23.05 release notes.
The removal won't be backported because it is considered a breaking change.
Please do not mention exact version numbers. They could be out of date on the 23.05 release.

File: nixos/doc/manual/release-notes/rl-2211.section.md
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2211.section.md:347]:
```suggestion
- The `aseprite-unfree` package has been upgraded from 1.2.16.3 to 1.2.40.
  The free version of aseprite has been dropped because it is EOL
  and the package attribute now points to the unfree version.
  A maintained fork of the last free version of Aseprite, named
  'LibreSprite', is available in the `libresprite` package.
```

File: pkgs/applications/editors/aseprite/default.nix
Comment (SuperSandro2000) [inline pkgs/applications/editors/aseprite/default.nix:1]:
```suggestion
{ stdenv, lib, callPackage, fetchFromGitHub, cmake, ninja, pkg-config
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by kyren** [comment]
Hey, I'm available to make any necessary changes, I didn't *completely* forget about this PR

**Comment by tjni** [comment]
Can you rebase, move the docs to the latest release per Sandro's suggestion, and squash the last two commits on this branch into your original second one?

I skimmed the changes again, and it's really impressive work, thank you for helping keep this package up to date and clean. I don't use `aseprite` so afterward I'll just double check that it builds and merge it. If anyone else on this thread has time and is able to do some testing, now's the time.

@orivej a courtesy ping in case you still would like to take a look.

**Comment by kyren** [comment]
> Can you rebase, move the docs to the latest release per Sandro's suggestion, and squash the last two commits on this branch into your original second one?

Done, but I'm not sure I understand the suggestion from SuperSandro2000:

>  Please do not mention exact version numbers. They could be out of date on the 23.05 release.

I've moved the release notes to 23.11 but I kept the precise version numbers (1.2.16.3 -> 1.2.40). It seems like every other entry in the release notes does the same thing? Maybe I misunderstood what was being said?

**Comment by tjni** [comment]
Cool, it LGTM. I was able to build it locally on an aarch64-linux VM as well. I think we can always make changes in future PRs if any are requested.

**Pull request merged**
